lib/Dialect/RTL/RTlOps.cpp

```diff
@@ -839,6 +839,23 @@ OpFoldResult ExtractOp::fold(ArrayRef<Attribute> operands) {
   return {};
 }
 
+static ParseResult parseSliceTypes(OpAsmParser &p, Type &srcType,
+                                   Type &idxType) {
+  if (p.parseType(srcType))
+    return failure();
+  auto srcArrayType = srcType.dyn_cast<ArrayType>();
+  if (!srcArrayType)
+    return failure();
```

Comment:
This need emit an error before returning failure.

Reply:
However, there is a better way: if you change the function signature to:

`static ParseResult parseSliceTypes(OpAsmParser &p, ArrayType &srcType,`

Then the parseType call will verify it is an ArrayType for you and print the error for you.  If you can't do this because of hte custom thing, then you can introduce a temporary ArrayType variable.